### PR TITLE
ci: add all builds tag

### LINKED ISSUE #1375: ci: add all builds tag
On addition of `ci: build all` label, ci should build the PR on all supported platforms.

**Technical Requirements**
- Reuse the matrix defined here: https://github.com/tailcallhq/tailcall/blob/main/.github/workflows/ci.yml#L181
- Use `labels.json` to create the new `ci: build matrix` label

### PR DESCRIPTION
**Pull request by Bhavyajain21**
/claim #1375 
/fixes #1375 

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
    - Added a feature to trigger builds on all supported platforms by using the new label "ci: build all."
    - Enhanced the CI workflow by introducing a new job that builds on various platforms based on pull request conditions and labels.

### CODE REVIEW

**Comment by Bhavyajain21** [comment]
Please review @tusharmath

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -498,3 +498,21 @@ jobs:
 
       - name: Update Homebrew Formula
         run: ./update-formula.sh ${{needs.draft_release.outputs.create_release_name }}
+
+  ci_build_all:
```
Comment (tusharmath) [inline .github/workflows/ci.yml:502]: WE need to reuse existing job and settings.

Comment (Bhavyajain21) [reply .github/workflows/ci.yml:502]: working on it

Comment (Bhavyajain21) [reply .github/workflows/ci.yml]: made the changes, please review @tusharmath

**Comment by Bhavyajain21** [comment]
@tusharmath can you please re-review this?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -111,27 +111,20 @@ jobs:
             files: lcov.info
             fail_ci_if_error: true
 
-  check-examples:
-    name: Check Examples
```
Comment (tusharmath) [inline .github/workflows/ci.yml:115]: did we delete this job?

**Comment by tusharmath** [comment]
@Bhavyajain21 Thanks for the change. Can you make a quick change here? Let's run the test all on platforms always, irrespective of the labels. 

Also you can use `&` to reduce duplication in YAMLs, check this cheatsheet out https://quickref.me/yaml.html

**Comment by Bhavyajain21** [comment]
> @Bhavyajain21 Thanks for the change. Can you make a quick change here? Let's run the test all on platforms always, irrespective of the labels.
> 
> Also you can use `&` to reduce duplication in YAMLs, check this cheatsheet out https://quickref.me/yaml.html

Hey @tusharmath , are we expecting a change anywhere else?

**Comment by Bhavyajain21** [comment]
> @Bhavyajain21 Thanks for the change. Can you make a quick change here? Let's run the test all on platforms always, irrespective of the labels.
> 
> Also you can use `&` to reduce duplication in YAMLs, check this cheatsheet out https://quickref.me/yaml.html

Got your point, making the change

**Comment by Bhavyajain21** [comment]
@tusharmath made the changes, please review.

**Comment by tusharmath** [comment]
@Bhavyajain21 This PR isn't making much sense anymore. What have you done? Labels change still exists.

Closing this PR for the sake of reducing noise. Feel free to create a new one whenever you think it's ready.

**Pull request closed without merging**